### PR TITLE
fix/tests time out

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: tests/e2e/fetchUserContacts.test.ts [hunk 1]
```diff
@@ -35,7 +38,7 @@ describe('web3mail.fetchMyContacts()', () => {
       data: { email: 'test2@gmail.com' },
       name: 'test do not use',
     });
-  }, 3 * MAX_EXPECTED_BLOCKTIME);
+  }, 2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME);
```
Comment (PierreJeanjacquot) [inline tests/e2e/fetchUserContacts.test.ts:41]: 2 tx => 2 * (1 block to submit tx + 1 block for confirmation) => 4
```suggestion
  }, 4 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME);
```

File: tests/e2e/sendEmail.test.ts [hunk 2]
```diff
@@ -44,13 +49,13 @@ describe('web3mail.sendEmail()', () => {
     });
     // avoid race condition with subgraph indexation
     await sleep(5_000);
-  }, 3 * MAX_EXPECTED_BLOCKTIME + 5_000);
+  }, 2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME + 5_000);
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:52]: 2 tx => 2 * (1 block to submit tx + 1 block for confirmation) => 4
```suggestion
  }, 4 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME + 5_000);
```

File: tests/e2e/sendEmail.test.ts [hunk 3]
```diff
@@ -44,13 +49,13 @@ describe('web3mail.sendEmail()', () => {
...
 
-  afterAll(async () => {
-    await dataProtector.revokeAllAccessObservable({
+  afterAll(() => {
+    dataProtector.revokeAllAccessObservable({
       protectedData: validProtectedData.address,
     });
-  }, 3 * MAX_EXPECTED_BLOCKTIME);
+  }, MAX_EXPECTED_WEB2_SERVICES_TIME);
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:58]: The cleanup is not triggered (subscribe is not called)
I would remove the whole `afterAll` block

File: tests/e2e/sendEmail.test.ts [hunk 4]
```diff
@@ -64,7 +69,7 @@ describe('web3mail.sendEmail()', () => {
       const sendEmailResponse = await web3mail.sendEmail(params);
       expect(sendEmailResponse.taskId).toBeDefined();
     },
-    3 * MAX_EXPECTED_BLOCKTIME
+    MAX_EXPECTED_WEB2_SERVICES_TIME
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:72]: `sendEmail` actually sends a `matchOrders` tx under the hood we must take this time into account.
```suggestion
    2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
```

File: tests/e2e/sendEmail.test.ts [hunk 5]
```diff
@@ -80,7 +85,7 @@ describe('web3mail.sendEmail()', () => {
...
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:88]: `sendEmail` actually sends a `matchOrders` tx under the hood we must take this time into account.
```suggestion
    2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
```

File: tests/e2e/sendEmail.test.ts [hunk 6]
```diff
@@ -115,7 +120,7 @@ describe('web3mail.sendEmail()', () => {
         'Dataset order not found'
       );
     },
-    3 * MAX_EXPECTED_BLOCKTIME
+    2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:123]: this test includes a 5s sleep time
```suggestion
    2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME + 5_000
```

File: tests/e2e/sendEmail.test.ts [hunk 7]
```diff
@@ -130,7 +135,7 @@ describe('web3mail.sendEmail()', () => {
       const sendEmailResponse = await web3mail.sendEmail(params);
       expect(sendEmailResponse.taskId).toBeDefined();
     },
-    3 * MAX_EXPECTED_BLOCKTIME
+    MAX_EXPECTED_WEB2_SERVICES_TIME
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:138]: `sendEmail` actually sends a `matchOrders` tx under the hood we must take this time into account.
```suggestion
    2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
```

File: tests/e2e/sendEmail.test.ts [hunk 8]
```diff
@@ -149,7 +154,7 @@ describe('web3mail.sendEmail()', () => {
...
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:157]: `sendEmail` actually sends a `matchOrders` tx under the hood we must take this time into account.
```suggestion
    2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
```

File: tests/e2e/sendEmail.test.ts [hunk 9]
```diff
@@ -212,9 +217,10 @@ describe('web3mail.sendEmail()', () => {
...
```
Comment (PierreJeanjacquot) [inline tests/e2e/sendEmail.test.ts:220]: `sendEmail` actually sends a `matchOrders` tx under the hood we must take this time into account.
```suggestion
    2 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
```

File: tests/e2e/sendEmail.test.ts (hunk 3 above)
Comment (PierreJeanjacquot) [reply tests/e2e/sendEmail.test.ts:58]: PR => https://github.com/iExecBlockchainComputing/web3mail-sdk/pull/105

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
- some timeout to changes see comments
- proposed cleanup https://github.com/iExecBlockchainComputing/web3mail-sdk/pull/105

File: tests/e2e/sendEmail.test.ts (hunk 3 above)
Comment (Le-Caignec) [reply tests/e2e/sendEmail.test.ts:58]: yeah i agree

**Review by PierreJeanjacquot — APPROVED** [review]

**Pull request merged**